packages/patterns/src/patterns/patternMatchers.js

```diff
@@ -1824,11 +1871,49 @@ const InterfaceGuardShape = M.splitRecord(
   },
 );
 
+/**
+ * @param {any} specimen
+ * @returns {asserts specimen is InterfaceGuard}
+ */
 export const assertInterfaceGuard = specimen => {
   mustMatch(specimen, InterfaceGuardShape, 'interfaceGuard');
 };
 harden(assertInterfaceGuard);
 
+/**
+ * By using this abstraction rather than accessing the properties directly,
+ * we smooth the transition to https://github.com/endojs/endo/pull/1712
+ *
+ * @param {InterfaceGuard} interfaceGuard
+ * @returns {InterfaceGuardPayload}
+ */
+export const getInterfaceGuardPayload = interfaceGuard => {
+  assertInterfaceGuard(interfaceGuard);
+  const { klass: _, ...payload } = interfaceGuard;
+  /** @type {InterfaceGuardPayload} */
+  // @ts-expect-error inference too weak to see this is ok.
+  return payload;
+};
+harden(getInterfaceGuardPayload);
+
+const emptyCopyMap = makeCopyMap([]);
+
+/**
+ * @param {InterfaceGuard} interfaceGuard
+ * @returns {(string | symbol)[]}
+ */
+export const getInterfaceMethodNames = interfaceGuard => {
```

Comment:
In ECMAScript, a property _name_ is always a string: https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-object-type
How closely should we hew to that vocabulary here?
```suggestion
export const getInterfaceMethodKeys = interfaceGuard => {
```